netwayste/src/utils/ping.rs

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2020 The Conwayste Developers
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use std::collections::VecDeque;
+use std::time::Instant;
+
+const PING_FILTER_DEPTH: usize = 12;
+
+pub struct PingFilter {
+    pub average_latency_ms: u128,
+    running_sum: u128,
+    history: VecDeque<u128>,
+    start_timestamp: Instant,
+    in_progress: bool,
+}
+
+impl PingFilter {
+    pub fn new() -> PingFilter {
+        PingFilter {
+            average_latency_ms: 0,
+            running_sum: 0,
+            history: VecDeque::with_capacity(PING_FILTER_DEPTH),
+            start_timestamp: Instant::now(),
+            in_progress: false,
+        }
+    }
+
+    pub fn reset(&mut self) {
+        let Self {
```

Comment:
You need a `#![deny(unused_variables)]` above this line for this pattern to be useful.

Reply:
Ah yes thank you!